### PR TITLE
Support ruby3

### PR DESCRIPTION
**Pull request by sachin21**
# WHAT
Supported ruby3.
Ruby3 String will not be able to use the `<<` without `.dup`.

# REF
- https://bugs.ruby-lang.org/issues/11473 (English)
- http://d.hatena.ne.jp/ku-ma-me/20151004/p1 (Japanese)

### CODE REVIEW

**Comment by sachin21** [comment]
@houndci Thank you for your comments :smile: 
I fixed issues of rubocop on this PR and squashed commits :+1: 
But I didn't resolve issues of Rubocop that exists already.

File: lib/paperclip/matchers/validate_attachment_content_type_matcher.rb [hunk 1]
```diff
@@ -65,7 +65,7 @@ def accepted_types_and_failures
         end
         def rejected_types_and_failures
           if @rejected_types.present?
-            "Reject content types: #{@rejected_types.join(", ")}\n".tap do |message|
+            "Reject content types: #{@rejected_types.join(", ")}\n".dup.tap do |message|
```
Comment (sachin21) [inline lib/paperclip/matchers/validate_attachment_content_type_matcher.rb]: @houndci I thought paperclip is not use the rubocop...
sorry. I fixed that

File: lib/paperclip/matchers/validate_attachment_content_type_matcher.rb [hunk 2]
```diff
@@ -54,7 +54,7 @@ def description
 
         def accepted_types_and_failures
           if @allowed_types.present?
-            "Accept content types: #{@allowed_types.join(", ")}\n".tap do |message|
+            "Accept content types: #{@allowed_types.join(", ")}\n".dup.tap do |message|
```
Comment (sachin21) [inline lib/paperclip/matchers/validate_attachment_content_type_matcher.rb]: too ref: https://github.com/thoughtbot/paperclip/pull/2033#discussion_r42825289

**Comment by sachin21** [comment]
@houndci Would you mind to merge this PR?

**Comment by sachin21** [comment]
omg
I found that out Mr. houndci is bot.
@tute How about this PR?

**Comment by tute** [comment]
Hey, thanks so much! But until we don't have Ruby 3 in CI, we can't, as we don't have regression tests. Does Travis already support Ruby 3?

**Comment by sachin21** [comment]
hmm, That is certainly true.
We'll wait once until it's supported. so May I stay this PR? until it's supported.
At that time remind to you :smile:

**Comment by tute** [comment]
Yep! Pending a version of ruby 3 in Travis. :+1:

**Comment by sachin21** [comment]
> Ruby 2.3 has a feature to help transition to the frozen string change of Ruby 3. Take a look at the section Frozen String Literals on http://nithinbekal.com/posts/ruby-2-3-features/. It looks like we can add a comment # frozen_string_literal: true to enable the frozen string functionality of Ruby 3. Ruby 2.3 does not have an official build yet so this may be a bit premature. I think we should add in the comment and fix the errors that occur because of the change.

https://github.com/bbatsov/rubocop/pull/2328#issuecomment-163628935

I think this idea is good! How about this latest commits?

@tute

**Comment by tute** [comment]
Thanks! Two questions:

* Can we enable this option automatically on every paperclip file, without adding a comment to each one of them?
* Instead of mutating the `path` string, can we build it in such a way that we can work on it as a frozen string?

Thanks again!

**Comment by sachin21** [comment]
> Can we enable this option automatically on every paperclip file, without adding a comment to each one of them?

No sir, We can't do that now. But when ruby3 is released, We can remove inline comment(for frozen string literal) in all *.rb files.
so let's wait the ruby3 version :smile: 
Next, 
> Instead of mutating the path string, can we build it in such a way that we can work on it as a frozen string?

Uhh, Do you mean this? https://github.com/sachin21/paperclip/commit/2174c915507587d8f7f9f74457c9d7fabd025201#diff-2590d28581541e7cccb81df6b758e191L5

Then, I’d like to propose this `# frozen_string_literal: false` and remove the dup method.
So Travis CI will be passed testing.

**Comment by bdewater** [comment]
Only string literals like `"foo"` are frozen by the comment. You could use `String.new("foo")` and mutate it. Call `#freeze` yourself if you want to have frozen strings everywhere.

**Comment by sachin21** [comment]
@bdewater ok I replaced to String.new instead of quotes :smile: 
and...
@tute how about [this](https://github.com/thoughtbot/paperclip/pull/2033#issuecomment-167507315) latest comment for u?

**Comment by tute** [comment]
> @tute how about this latest comment for u?

Wait for a version of ruby 3 makes sense to me, so we can test without the comments, and current solution for the path string (without dup) too.

Thank you!

**Comment by sachin21** [comment]
I understand. I'll close this pull-request :smile:

**Pull request closed without merging**